File: Domain/Dtos/VehicleRequest.cs
namespace Domain.Dtos;

public record VehicleRequest(
    int MakeId,
    int ModelId,
    string? VIN,
    string? LicensePlate,
    int Year,
    DateOnly? PurchaseDate,
    decimal? PurchasePrice,
    int? MileageAtPurchase,
    int? Mileage
);

File: frontend/src/types/vehicleRequest.ts
export type VehicleRequest = {
  makeId: number;
  modelId: number;
  year: number;
  vin?: string | null;
  licensePlate?: string | null;
  purchaseDate?: string | null;
  purchasePrice?: number | null;
  mileageAtPurchase?: number | null;
  mileage?: number | null;
};
